tests/functional/idempotency/test_idempotency.py

```diff
@@ -845,3 +848,41 @@ def handler(event, context):
         handler({}, lambda_context)
 
     assert "No data found to create a hashed idempotency_key" == e.value.args[0]
+
+
+class MockPersistenceLayer(BasePersistenceLayer):
+    def __init__(self, expected_idempotency_key: str):
+        self.expected_idempotency_key = expected_idempotency_key
+        super(MockPersistenceLayer, self).__init__()
+
+    def _put_record(self, data_record: DataRecord) -> None:
+        assert data_record.idempotency_key == self.expected_idempotency_key
+
+    def _update_record(self, data_record: DataRecord) -> None:
+        assert data_record.idempotency_key == self.expected_idempotency_key
+
+    def _get_record(self, idempotency_key) -> DataRecord:
+        ...
+
+    def _delete_record(self, data_record: DataRecord) -> None:
+        ...
```

Comment:
I'm on the fence about this one - I might remove it in the future if we end up not reusing it elsewhere in the tests.

Reply:
We could use it to verify if the parser works with idempotent decorator.